File: android/app/src/main/java/com/scanx/app/data/PageFilter.kt
package com.scanx.app.data

/**
 * Bộ lọc xem/xuất RIÊNG TỪNG TRANG (bản 0.8, màn hình "Chỉnh sửa trang" → tab "Bộ lọc") — độc lập
 * với cỡ file (nhỏ gọn/chất lượng cao) chọn lúc xuất cả tài liệu ([PdfExportMode]). Trang không đặt
 * (null) → dùng đúng theo [PdfExportMode] của tài liệu như trước bản 0.8 (không đổi hành vi cũ);
 * trang có đặt → override phần "màu hay đen trắng/xám" của riêng trang đó, không ảnh hưởng trang khác.
 */
enum class PageFilter(val code: String, val label: String) {
    ORIGINAL("ORIGINAL", "Ban đầu"),
    AUTO("AUTO", "Tự động"),
    COLOR("COLOR", "Màu"),
    BW("BW", "Đen trắng"),
    GRAY("GRAY", "Xám"),
    SHADOW("SHADOW", "Bóng"),
    ;

    companion object {
        fun fromCode(code: String?): PageFilter? = values().firstOrNull { it.code == code }
    }
}

File: ios/ScanX/PageFilter.swift
import Foundation

/// Bộ lọc riêng từng trang — cùng mã với Android (data/PageFilter.kt). Lưu ở meta.pageFilters,
/// không nướng vào ảnh gốc; nil = trang theo chế độ PDF của tài liệu.
enum PageFilter: String, CaseIterable, Identifiable, Codable {
    case original = "ORIGINAL"
    case auto = "AUTO"
    case color = "COLOR"
    case bw = "BW"
    case gray = "GRAY"
    case shadow = "SHADOW"

    var id: String { rawValue }

    var title: String {
        switch self {
        case .original: return "Ban đầu"
        case .auto: return "Tự động"
        case .color: return "Màu"
        case .bw: return "Đen trắng"
        case .gray: return "Xám"
        case .shadow: return "Bóng"
        }
    }
}
